dlt/destinations/job_client_impl.py

```diff
@@ -404,18 +405,20 @@ def _update_schema_in_storage(self, schema: Schema) -> None:
         version_hash = schema.version_hash
         if version_hash != schema.stored_version_hash:
             raise DestinationSchemaTampered(schema.name, version_hash, schema.stored_version_hash)
-        now_ts = str(pendulum.now())
         # get schema string or zip
         schema_str = json.dumps(schema.to_dict())
         # TODO: not all databases store data as utf-8 but this exception is mostly for redshift
         schema_bytes = schema_str.encode("utf-8")
         if len(schema_bytes) > self.capabilities.max_text_data_type_length:
             # compress and to base64
             schema_str = base64.b64encode(zlib.compress(schema_bytes, level=9)).decode("ascii")
+        self._commit_schema_update(schema, schema_str)
+
+    def _commit_schema_update(self, schema: Schema, schema_str: str) -> None:
         # insert
         name = self.sql_client.make_qualified_table_name(VERSION_TABLE_NAME)
+        now_ts = str(pendulum.now())
         # values =  schema.version_hash, schema.name, schema.version, schema.ENGINE_VERSION, str(now_ts), schema_str
         self.sql_client.execute_sql(
             f"INSERT INTO {name}({self.VERSION_TABLE_SCHEMA_COLUMNS}) VALUES (%s, %s, %s, %s, %s, %s);", schema.stored_version_hash, schema.name, schema.version, schema.ENGINE_VERSION, now_ts, schema_str
```

Comment:
this must work, converting to dict style should be easy, see above

Reply:
it does, I had to add a custom formatter but this is quite nice now